File: Services/FolderService.swift
import Foundation

/// 文件夹业务操作服务
final class FolderService: @unchecked Sendable {
    private let folderRepo: FolderRepository
    private let itemRepo: ItemRepository

    init(
        folderRepo: FolderRepository = FolderRepository(),
        itemRepo: ItemRepository = ItemRepository()
    ) {
        self.folderRepo = folderRepo
        self.itemRepo = itemRepo
    }

    /// 删除文件夹：先将 folder 内所有 items 的 folderID 清空，再删除文件夹本身
    func deleteFolder(id: UUID) throws {
        let itemsInFolder = (try? itemRepo.fetchAll(folderID: id)) ?? []
        for var item in itemsInFolder {
            item.folderID = nil
            try? itemRepo.update(item)
        }
        try folderRepo.delete(id: id)
    }
}

File: windows/src/Gatherly.Windows/Services/FolderService.cs
using Gatherly.Windows.Database;
using Gatherly.Windows.Models;
using Gatherly.Windows.Models.Enums;

namespace Gatherly.Windows.Services;

/// <summary>
/// 文件夹业务服务 — 对齐 macOS FolderService
/// </summary>
public class FolderService
{
    private readonly FolderRepository _folderRepo;
    private readonly ItemRepository _itemRepo;

    public FolderService(FolderRepository folderRepo, ItemRepository itemRepo)
    {
        _folderRepo = folderRepo;
        _itemRepo = itemRepo;
    }

    public async Task<Folder> CreateFolderAsync(string name, Platform platform,
        Guid? parentPlatformId = null, Guid? customPlatformId = null)
    {
        if (string.IsNullOrWhiteSpace(name))
            throw new ArgumentException("文件夹名称不能为空");

        return await _folderRepo.CreateAsync(name.Trim(), platform, parentPlatformId, customPlatformId);
    }

    public async Task RenameFolderAsync(Guid folderId, string newName)
    {
        if (string.IsNullOrWhiteSpace(newName))
            throw new ArgumentException("文件夹名称不能为空");

        var existing = await _folderRepo.GetByIdAsync(folderId)
            ?? throw new InvalidOperationException("文件夹不存在");

        await _folderRepo.UpdateNameAsync(folderId, newName.Trim());
    }

    public async Task DeleteFolderAsync(Guid folderId)
    {
        var existing = await _folderRepo.GetByIdAsync(folderId)
            ?? throw new InvalidOperationException("文件夹不存在");

        // 对齐 macOS：清空文件夹内所有 items 的 folderID，但不删除 items
        var itemsInFolder = await _itemRepo.GetByFolderIdAsync(folderId);
        foreach (var item in itemsInFolder)
        {
            item.FolderId = null;
            item.ModifyDate = DateTimeOffset.UtcNow;
            await _itemRepo.UpdateAsync(item);
        }

        await _folderRepo.DeleteAsync(folderId);
    }

    public async Task<int> CountItemsAsync(Guid folderId)
    {
        return await _folderRepo.CountItemsAsync(folderId);
    }
}
